pkg/manifests/config.go

```diff
@@ -126,6 +127,27 @@ func (c Config) GetThanosRulerAlertmanagerConfigs() []AdditionalAlertmanagerConf
 	return alertmanagerConfigs
 }
 
+func (c Config) HasInvalidScrapeIntervalDuration() bool {
+	if c.ClusterMonitoringConfiguration == nil || c.UserWorkloadConfiguration == nil {
+		return false
+	}
+
+	if c.UserWorkloadConfiguration.Prometheus == nil || c.UserWorkloadConfiguration.Prometheus.ScrapeInterval == "" {
+		return false
+	}
+
+	scrapeInterval, err := model.ParseDuration(c.UserWorkloadConfiguration.Prometheus.ScrapeInterval)
+
+	if err != nil {
+		return true
+	}
+
+	allowedLowerLimit, _ := model.ParseDuration("5s")
+	allowedUpperLimit, _ := model.ParseDuration("5m")
+
+	return (scrapeInterval < allowedLowerLimit) || (scrapeInterval > allowedUpperLimit)
```

Comment:
It's better user experience to report why the value isn't valid.
```suggestion
	if (scrapeInterval < allowedLowerLimit) || (scrapeInterval > allowedUpperLimit) {
		return fmt.Errorf("scrape interval value %q outside of the allowed range [%q, %q]", c.UserWorkloadConfiguration.Prometheus.ScrapeInterval, allowedLowerLimit, allowedUpperLimit)
```